### PR TITLE
cmake: Fix diverging prefix padding for OBS status outputs

### PR DESCRIPTION
**Pull request by PatTheMav**
### Description
Status output related to OBS configuration is prefixed with the string
"OBS" and added padding for enabled and disabled features. This padding
was not aligned between platforms.

### Motivation and Context
By moving the padding and prefix decoration into its own function,
both elements are controlled in a single place. CMake scripts were
changed to use this new function `obs_status` instead of using CMake's
`message` function directly.

### How Has This Been Tested?
* Tested with local builds on Windows, macOS and Linux

### Types of changes
- Tweak (non-breaking change to improve existing functionality)

### Checklist:
- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Comment by RytoEX** [comment]
Technically, the prefix is "cmake: " (all lowercase, as the folder name). This ensures that case-sensitive searches, such as git-gui's default search method, match a search for "cmake".

**Review by RytoEX — COMMENTED** [review]
Just one clarifying question.

File: deps/obs-scripting/CMakeLists.txt [hunk 1]
```diff
@@ -13,13 +13,13 @@ if(ENABLE_SCRIPTING_LUA)
   find_package(Luajit)
 
   if(NOT TARGET Luajit::Luajit)
-    message(FATAL_ERROR "OBS:    -        Luajit not found")
+    obs_status(FATAL_ERROR "obs-scripting -> Luajit not found.")
```
[the remark this answers is not in the document]
Comment (PatTheMav) [reply deps/obs-scripting/CMakeLists.txt:16]: That and a stylistic choice.

[the remark this answers is not in the document]
Comment (RytoEX) [reply deps/obs-scripting/CMakeLists.txt:16]: Fair enough.

**Pull request merged**